storage/catalog/fetcher_test.go

```diff
@@ -0,0 +1,172 @@
+/*
+ * Copyright 2018 The Service Manager Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package catalog_test
+
+import (
+	"context"
+	"encoding/base64"
+	"fmt"
+	"net/http"
+
+	"github.com/Peripli/service-manager/pkg/util"
+
+	. "github.com/onsi/gomega"
+
+	"github.com/Peripli/service-manager/storage/catalog"
+
+	"github.com/Peripli/service-manager/pkg/types"
+	"github.com/Peripli/service-manager/test/common"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/ginkgo/extensions/table"
+)
+
+var _ = Describe("Catalog Fetcher", func() {
+	const (
+		simpleCatalog = `
+		{
+		  "services": [{
+				"name": "no-tags-no-metadata",
+				"id": "acb56d7c-XXXX-XXXX-XXXX-feb140a59a67",
+				"description": "A fake service.",
+				"dashboard_client": {
+					"id": "id",
+					"secret": "secret",
+					"redirect_uri": "redirect_uri"		
+				},    
+				"plans": [{
+					"random_extension": "random_extension",
+					"name": "fake-plan-1",
+					"id": "d3031751-XXXX-XXXX-XXXX-a42377d33202",
+					"description": "Shared fake Server, 5tb persistent disk, 40 max concurrent connections.",
+					"free": false
+				}]
+			}]
+		}`
+
+		id       = "id"
+		name     = "name"
+		url      = "url"
+		username = "username"
+		password = "password"
+		version  = "2.13"
+	)
+
+	var testBroker *types.ServiceBroker
+
+	type testCase struct {
+		expectations *common.HTTPExpectations
+		reaction     *common.HTTPReaction
+
+		expectedErr      error
+		expectedResponse []byte
+	}
+
+	newFetcher := func(t testCase) func(ctx context.Context, broker *types.ServiceBroker) ([]byte, error) {
+		return catalog.Fetcher(common.DoHTTP(t.reaction, t.expectations), version)
+	}
+
+	basicAuth := func(username, password string) string {
+		auth := username + ":" + password
+		return base64.StdEncoding.EncodeToString([]byte(auth))
+	}
+
+	BeforeEach(func() {
+		testBroker = &types.ServiceBroker{
```

Comment:
There's no need to construct the broker in a `BeforeEach` every single time

Reply:
There is also no way in go to declare it as a constant - this is the only way to guarantee it will be in the "expected" state before each test runs - otherwise someone might add more tests later on that modify it and this would lead to tests not testing what should be tested

Reply:
Clever, I understand